youtube-client/src/app/auth/pages/admin/admin.component.ts

```diff
@@ -0,0 +1,34 @@
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-admin',
+  templateUrl: './admin.component.html',
+  styleUrls: ['./admin.component.scss'],
+})
+export default class AdminComponent implements OnInit {
+  adminForm: FormGroup;
+
+  ngOnInit(): void {
+    this.adminForm = new FormGroup({
+      title: new FormControl('', [Validators.required, Validators.minLength(3), Validators.maxLength(20)]),
+      description: new FormControl('', Validators.maxLength(255)),
+      imageUrl: new FormControl('', [Validators.required, Validators.pattern(/https:\/\//)]),
+      videoLink: new FormControl('', [Validators.required, Validators.pattern(/https:\/\//)]),
+      date: new FormControl('', [Validators.required, this.customDateValidators]),
+    });
+  }
+
+  private customDateValidators(control: FormControl): null | { [key: string]: boolean } {
```

Comment:
This doesn't need to be a part of the component. Either move it outside as a const, or create a service that exposes this method